File: android/plugins/com-omariskandarani-vlcremote-plugin/ios/Plugin/getJsonPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(getJsonPlugin)
public class getJsonPlugin: CAPPlugin {
    private let implementation = getJson()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/plugins/com-omariskandarani-vlcremote-plugin/android/src/main/java/com/omariskandarani/vlcremote/plugin/getJsonPlugin.java
package com.omariskandarani.vlcremote.plugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "getJson")
public class getJsonPlugin extends Plugin {

    private getJson implementation = new getJson();

    @PluginMethod
    public void echo(PluginCall call) {
        String pass = call.getString("pass", "");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(pass));
        call.resolve(ret);
    }
}
